src/main/java/com/github/princesslana/eriscasper/rest/GuildRoute.java

```diff
@@ -11,22 +40,323 @@ private GuildRoute(Snowflake id) {
+   *     https://discordapp.com/developers/docs/resources/guild#modify-guild-member</a>
+   */
+  public Route<ModifyGuildMemberRequest, Void> modifyGuildMember(Snowflake userId) {
+    return Route.patch(
+        path("/members/%s", userId.unwrap()), ModifyGuildMemberRequest.class, Void.class);
+  }
+
+  /**
+   * @see <a href="https://discordapp.com/developers/docs/resources/guild#modify-current-user-nick">
+   *     https://discordapp.com/developers/docs/resources/guild#modify-current-user-nick</a>
+   */
+  public Route<ModifyCurrentNickRequest, String> modifyCurrentUserNick() {
+    return Route.patch(path("/members/@me/nick"), ModifyCurrentNickRequest.class, String.class);
+  }
+
   /**
    * @see <a href="https://discordapp.com/developers/docs/resources/guild#add-guild-member-role">
    *     https://discordapp.com/developers/docs/resources/guild#add-guild-member-role</a>
    */
-  public Route<Void, Void> addGuildMemberRole(Snowflake userId, Snowflake guildId) {
+  public Route<Void, Void> addGuildMemberRole(Snowflake userId, Snowflake roleId) {
     return Route.put(
-        path("/members/%s/roles/%s", userId.unwrap(), guildId.unwrap()), Void.class, Void.class);
+        path("/members/%s/roles/%s", userId.unwrap(), roleId.unwrap()), Void.class, Void.class);
   }
 
   /**
    * @see <a href="https://discordapp.com/developers/docs/resources/guild#remove-guild-member-role">
    *     https://discordapp.com/developers/docs/resources/guild#remove-guild-member-role</a>
    */
-  public Route<Void, Void> removeGuildMemberRole(Snowflake userId, Snowflake guildId) {
-    return Route.delete(
-        path("/members/%s/roles/%s", userId.unwrap(), guildId.unwrap()), Void.class);
+  public Route<Void, Void> removeGuildMemberRole(Snowflake userId, Snowflake roleId) {
+    return Route.delete(path("/members/%s/roles/%s", userId.unwrap(), roleId.unwrap()), Void.class);
+  }
+
+  /**
+   * @see <a href="https://discordapp.com/developers/docs/resources/guild#remove-guild-member">
+   *     https://discordapp.com/developers/docs/resources/guild#remove-guild-member</a>
+   */
+  public Route<Void, Void> removeGuildMember(Snowflake userId) {
+    return Route.delete(path("/members/%s", userId.unwrap()), Void.class);
+  }
+
+  /**
+   * @see <a href="https://discordapp.com/developers/docs/resources/guild#get-guild-bans">
+   *     https://discordapp.com/developers/docs/resources/guild#get-guild-bans</a>
+   */
+  public Route<Void, ImmutableList<Ban>> getGuildBans() {
+    return Route.get(path("/bans"), Route.jsonArrayResponse(Ban.class));
+  }
+
+  /**
+   * @see <a href="https://discordapp.com/developers/docs/resources/guild#get-guild-ban">
+   *     https://discordapp.com/developers/docs/resources/guild#get-guild-ban</a>
+   */
+  public Route<Void, Optional<Ban>> getGuildBan(Snowflake userId) {
+    return Route.get(
+        path("/bans/%s" + userId.unwrap()),
+        (response) ->
+            response.code() == 404
```

Comment:
We shouldn't be dealing with 404 here. Most likely this throws an exception when the response is received anyway.

Reply:
No, it should be a clean 404 according to documentation.

Reply:
Anything other than a 2xx response is treated as an error at the time we get the http response.

Handling this in a way other than an exception is a job for layers of code above this.

Reply:
We should handle it above then because we are expecting a 404 when there's no ban set for a user.

Reply:
Sure, we're expecting 404 in a number of places I imagine (e.g., if they try an get a guild with an id that doesn't exist)

Reply:
Alright, makes sense, they are able to get all the bans, so it might not be important to watch for it here.